### PR TITLE
[Design2-142] Design System - Breadcrumbs Component

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design2-142] Design System - Breadcrumbs Component

* implemented component
* added tests
* added storybook
* component is now ```Ready for Dev```

https://github.com/user-attachments/assets/7c23d691-fc1b-4b63-bcb3-1846d01f7889

## Links
- jira ticket: [Design2-142](https://codedotorg.atlassian.net/browse/DESIGN2-142)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx [hunk 1]
```diff
@@ -1,120 +1,70 @@
...
-
-import BreadCrumb from './_Breadcrumb';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import Link, {LinkBaseProps} from '@cdo/apps/componentLibrary/link';
 
 import moduleStyles from './breadcrumbs.module.scss';
 
+interface BreadcrumbProps extends LinkBaseProps {
+  text: string;
+  href: string;
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx]: `href` is already defined on `LinkBaseProps` right?

Comment (sanchitmalhotra126) [reply apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx]: Also if `text` is required, should just extend/be `LinkWithText`?

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM! Just a few comments on props/interfaces

File: apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx]: Yes, thanks for noticing!
I added this initially for better documentation, but it looks like it makes no significant difference to create a separate type here, will allow `LinkWithText` type import in a separate PR and will use it here

Comment (levadadenys) [reply apps/src/componentLibrary/breadcrumbs/Breadcrumbs.tsx]: https://github.com/code-dot-org/code-dot-org/pull/61281

**Pull request merged**